pkg/kubectl/cmd/util/factory.go

```diff
@@ -597,21 +600,57 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 	}
 }
 
-// GetFirstPod returns the first pod of an object from its namespace and selector and the number of matching pods
-func GetFirstPod(client *client.Client, namespace string, selector labels.Selector) (*api.Pod, int, error) {
-	var pods *api.PodList
-	for pods == nil || len(pods.Items) == 0 {
-		var err error
-		options := api.ListOptions{LabelSelector: selector}
-		if pods, err = client.Pods(namespace).List(options); err != nil {
-			return nil, 0, err
+// GetFirstPod returns a pod matching the namespace and label selector
+// and the number of all pods that match the label selector. reverse in
+// this context is the direction that the pods will be sorted. true means
+// they will be sorted from healthy to unhealthy.
+func GetFirstPod(client *client.Client, namespace string, selector labels.Selector, reverse bool) (*api.Pod, int, error) {
```

Comment:
How about we do `sort.Interface` instead of `bool` so we can drop in whatever we want?